### ISSUE TITLE
[DependencyInjection] Unable to create service definition with nested definition argument with private alias dependency

### ISSUE THREAD

**Issue by scaytrase** [symfony/symfony#24883]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.8.28

Background:

We have DIC extension which configures some services with `Definition` arguments using `methodCall`. Inside this definitions we use `Reference`s to another services. Everything works fine until the reference is  not a private alias (public works fine too)

Test case to reproduce

```php
use PHPUnit\Framework\TestCase;
use Symfony\Component\DependencyInjection\Alias;
use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\DependencyInjection\Definition;
use Symfony\Component\DependencyInjection\Reference;

class ContainerTest extends TestCase
{
    public function getAliasStatus()
    {
        return [
            'true' => [true],
            'false' => [false],
        ];
    }

    /**
     * @dataProvider getAliasStatus
     *
     * @param bool $public
     */
    public function testDefinitionReferencesAlias(bool $public)
    {
        $container = new ContainerBuilder();

        $container->register('original_dep', \stdClass::class);
        $container->setAlias('dependency', new Alias('original_dep', $public));

        $container->register('service', \stdClass::class)
            ->setArguments(
                [
                    (new Definition(\stdClass::class))
                        ->setArguments(
                            [
                                new Reference('dependency'),
                            ]
                        ),
                ]
            );

        $container->compile();
        $container->get('service');
    }

    public function testDefinitionReferencesService()
    {
        $container = new ContainerBuilder();

        $container->register('original_dep', \stdClass::class);
        $container->setAlias('dependency', new Alias('original_dep', false));

        $container->register('service', \stdClass::class)
            ->setArguments(
                [
                    (new Definition(\stdClass::class))
                        ->setArguments(
                            [
                                new Reference('original_dep'),
                            ]
                        ),
                ]
            );

        $container->compile();
        $container->get('service');
    }
}
```

Test case `testDefinitionReferencesAlias` with `public=false` fails with symfony 2.8.28

Investigations:
* `ResolveReferencesToAliasesPass` is ignored when argument is `Definition`
* `Alias` stand private there
* Private alias gets removed
* But reference to this alias is still in place, so exception is thrown

Suggestion:
* extract these lines to private method `processDefinition` https://github.com/symfony/dependency-injection/blob/bc845111480786a9a68b39578ecdf27d9a6a44ec/Compiler/ResolveReferencesToAliasesPass.php#L40-L44
* add one more `elseif` branch here https://github.com/symfony/dependency-injection/blob/2.8/Compiler/ResolveReferencesToAliasesPass.php#L72 for `Definition` argument, calling this method

Or maybe I just go the wrong way

**Comment by scaytrase** [comment]
Created  PR #24889 with these test cases and proposed fix

**Comment by nicolas-grekas** [comment]
This has been fixed on 3.4, and considered as a new feature. Therefore, I'm closing.
